src/connector/src/parser/avro/schema_resolver.rs

```diff
@@ -50,23 +56,17 @@ impl ConfluentSchemaCache {
         }
     }
 
-    pub async fn get_by_subject_name(&self, subject_name: &str) -> ConnectorResult<Arc<Schema>> {
-        let raw_schema = self.get_raw_schema_by_subject_name(subject_name).await?;
-        self.parse_and_cache_schema(raw_schema).await
-    }
-
-    pub async fn get_raw_schema_by_subject_name(
-        &self,
-        subject_name: &str,
-    ) -> ConnectorResult<ConfluentSchema> {
-        self.confluent_client
+    /// Gets the latest schema by name, which is used as *reader schema*.
+    pub async fn get_by_name(&self, subject_name: &str) -> ConnectorResult<Arc<Schema>> {
```

Comment:
nit: what about calling it `get_by_subject`? Because there are actually 3 concepts here: subject name, record name and topic name. The first is derived from the latter two based on selected subject naming strategy.